### PR TITLE
Support `alternate_id` for PO creation and lookup

### PR DESCRIPTION
**Pull request by dnewh**
This updates the Grid PO store, SDE, and CLI to support PO creation and lookup using a properly formatted (`<id_type>:<id>`) alternate ID. This change includes an update to the `grid create po` CLI command to update the flag for specifying an alternate ID to `--alternate-id` from the less descriptive `--id`.

### CODE REVIEW

File: cli/man/grid-po-create.1.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ output.
 OPTIONS
 =======
 
-`--id`
```
Comment (leebradley) [inline cli/man/grid-po-create.1.md:45]: We should update the example on line 86 to also use `--alternate-id`.

**Review by leebradley — CHANGES_REQUESTED** [review]
Tested locally. One minor change, otherwise LGTM.

File: cli/man/grid-po-create.1.md (hunk 1 above)
Comment (dnewh) [reply cli/man/grid-po-create.1.md:45]: updated

**Review by leebradley — APPROVED** [review]

**Review by shannynalayna — APPROVED** [review]
Tested

**Review by agunde406 — APPROVED** [review]

**Pull request merged**